swap/protocol.go

```diff
@@ -53,23 +49,26 @@ func (s *Swap) APIs() []rpc.API {
 		{
 			Namespace: "swap",
 			Version:   "1.0",
-			Service:   s.Service,
+			Service:   s.api,
 			Public:    false,
 		},
 	}
 }
 
-func (ss *Service) Start(server *p2p.Server) error {
+func (s *Swap) Start(server *p2p.Server) error {
 	log.Info("Swap service started")
 	return nil
 }
 
-func (ss *Service) Stop() error {
+func (s *Swap) Stop() error {
 	return nil
 }
 
-func (ss *Service) run(p *p2p.Peer, rw p2p.MsgReadWriter) error {
+func (s *Swap) run(p *p2p.Peer, rw p2p.MsgReadWriter) error {
 	protoPeer := protocols.NewPeer(p, rw, Spec)
-	swapPeer := NewPeer(protoPeer, ss.swap)
+	swapPeer := NewPeer(protoPeer, s, s.backend)
 	return swapPeer.Run(swapPeer.handleMsg)
 }
+
+type PublicAPI struct {
```

Comment:
Is this used somewhere?

Reply:
This is supposed to hold the handlers for the public API endpoints, similar to `Inspector` struct, within the `bzz` namespace. I am not sure if it is `required`, or `optional`, but this is why we added it.